### DISCUSSION TITLE
Getting rid of user warning

### DISCUSSION THREAD

**Discussion by derrickcchow** [vaexio/vaex#1861] [Q&A]
Hi,

When, I rename a column name. I am getting a UserWarning: The state wants to rename Partial Period to __Partial Period, but __Partial Period was not found, ignoring the rename.  Is there a way to get rid of this error or even silence it?

Thanks, Derrick

**Comment by JovanVeljanoski** [comment]
Can you please give us minimal example? Also which vaex version are you using?

**Reply by derrickcchow** [reply]
vaex version: vaex~=4.7.0
I am using vaex with plotly dash components.

The column data type starts as a string.
df_vaex['Fiscal Week of Event'] = df_vaex.func.where(df_vaex['Fiscal Week of Event'] == "", np.nan, df_vaex['Fiscal Week of Event'])
df_vaex['Fiscal Week of Event'] = df_vaex['Fiscal Week of Event'].astype('float64')

[2022-01-26 15:04:23,022] INFO in _internal.py (fn:_log ln:113): 127.0.0.1 - - [26/Jan/2022 15:04:23] "POST /python/dashboard/_dash-update-component HTTP/1.1" 200 -
[2022-01-26 15:04:26,565] ERROR in config.py (fn:write ln:74): C:\Users\derek\AppData\Local\Programs\Python\Python39\lib\site-packages\vaex\dataframe.py:2628: UserWarning: The state wants to rename Week of Event to __Week of Event, but __Week of Event was not found, ignoring the rename
[2022-01-26 15:04:26,565] ERROR in config.py (fn:write ln:74):   warnings.warn(f'The state wants to rename {old} to {new}, but {new} was not found, ignoring the rename')

**Comment by JovanVeljanoski** [comment]
I thought of a minimal example that we can run to reproduce your issue. 
For instance I tried this:

```python
import vaex
import numpy as np

df = vaex.from_arrays(x=['', '', '1', '2', '3'])
df.variables['nan'] = np.nan
df['x'] = df.func.where(df.x=='', np.nan, df.x)
df['x'] = df['x'].astype('float64')
```

And I don't get that user warning. Btw, are you getting just a warning or an actual error? 

Another note:
This like `df_vaex.func.where(df_vaex['Fiscal Week of Event'] == "", np.nan, df_vaex['Fiscal Week of Event'])` leads to a dtype object, and is probably a bad idea.. 

@maartenbreddels maybe we should have a way to force-cast an expression to a numberic one, that will make all non-numeric elements `np.nan`. Basically similar to `pd.to_numeric`. There was also a request in the issues somewhere at some point. 
What do you think?


Oh I found the issue, also @derrickcchow maybe this will be helpful to you too:
https://github.com/vaexio/vaex/issues/1331

**Reply by derrickcchow** [reply]
It's more of an error message in the terminal!
